apps/api/src/app/workflows-v2/usecases/build-step-issues/build-step-issues.usecase.ts

```diff
@@ -0,0 +1,258 @@
+import { Injectable } from '@nestjs/common';
+import { ControlValuesRepository } from '@novu/dal';
+import {
+  ContentIssue,
+  JSONSchemaDto,
+  StepContentIssueEnum,
+  StepIssuesDto,
+  UserSessionData,
+  StepTypeEnum,
+  WorkflowOriginEnum,
+  ControlValuesLevelEnum,
+} from '@novu/shared';
+import {
+  InstrumentUsecase,
+  TierRestrictionsValidateUsecase,
+  TierRestrictionsValidateCommand,
+  dashboardSanitizeControlValues,
+  PinoLogger,
+} from '@novu/application-generic';
+
+import _ from 'lodash';
+import Ajv, { ErrorObject } from 'ajv';
+import addFormats from 'ajv-formats';
+import { buildVariables } from '../../util/build-variables';
+import { BuildAvailableVariableSchemaCommand, BuildAvailableVariableSchemaUsecase } from '../build-variable-schema';
+import { BuildStepIssuesCommand } from './build-step-issues.command';
+
+@Injectable()
+export class BuildStepIssuesUsecase {
+  constructor(
+    private buildAvailableVariableSchemaUsecase: BuildAvailableVariableSchemaUsecase,
+    private controlValuesRepository: ControlValuesRepository,
+    private tierRestrictionsValidateUsecase: TierRestrictionsValidateUsecase,
+    private logger: PinoLogger
+  ) {}
+
+  @InstrumentUsecase()
+  async execute(command: BuildStepIssuesCommand): Promise<StepIssuesDto> {
+    const {
+      workflowOrigin,
+      user,
+      step: foundPersistedStep,
+      workflow: persistedWorkflow,
+      stepDto: step,
+      controlSchema,
+    } = command;
+
+    const variableSchema = await this.buildAvailableVariableSchemaUsecase.execute(
+      BuildAvailableVariableSchemaCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        stepInternalId: foundPersistedStep?.stepId,
+        workflow: persistedWorkflow,
+        ...(step.controlValues ? { optimisticControlValues: step.controlValues } : {}),
+      })
+    );
+
+    let controlValueLocal = step.controlValues;
+
+    if (!controlValueLocal) {
+      controlValueLocal = (
```

Comment:
❓ Since issues are persisted in the workflow, shouldn't we compute them only on the provided control values (the one the frontend just sent)? 

Is there a use case in that we need to re-sanitize and stored control values?

Reply:
As decided previously, we **never** store `sanitizedControlValues`. Instead, we store the control values exactly as they are received from the front-end. Because of this, we need to sanitize the control values every time they are used, including in the following cases:  

- Issue generation  
- Preview  
- Trigger  

Hope it clarifies the question, let me know if additional information is needed :)

Reply:
It does. Thank you.